compiler-core/src/javascript/tests/case.rs

```diff
@@ -413,3 +413,59 @@ pub fn go(x) {
 "
     );
 }
+
+#[test]
+fn bit_array_assignment_utf16() {
+    assert_js!(
+        r#"
+pub fn main() {
+  case <<"Hello":utf16>> {
+    <<"Hello" as m:utf16>> -> m
+    _ -> ""
+  }
+}
+"#
+    );
+}
+
+#[test]
+fn bit_array_assignment_utf32() {
+    assert_js!(
+        r#"
+pub fn main() {
+  case <<"Hello":utf32>> {
+    <<"Hello" as m:utf32>> -> m
+    _ -> ""
+  }
+}
+"#
+    );
+}
+
+#[test]
+fn bit_array_assignment_utf16_little_endian() {
+    assert_js!(
+        r#"
+pub fn main() {
+  case <<"Hello":utf16-little>> {
+    <<"Hello" as m:utf16-little>> -> m
+    _ -> ""
+  }
+}
+"#
+    );
+}
+
+#[test]
+fn bit_array_assignment_utf32_little_endian() {
+    assert_js!(
+        r#"
+pub fn main() {
+  case <<"Hello":utf32-little>> {
+    <<"Hello" as m:utf32-little>> -> m
+    _ -> ""
+  }
+}
+"#
+    );
```

Comment:
Could we also have a test with an alternative pattern assigning a literal string with two different encodings?
```
<<"wibble" as m:utf16, _:bits>>
| <<"wobble" as m:utf32, 1, _:bits>> -> ...
```